### ISSUE TITLE
Incorrect rate limiting for some resources

### ISSUE THREAD

**Issue by clementdenis** [okta/terraform-provider-okta#1417]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Configuration Files

Any configuration using max_api_capacity

### Expected Behavior

The rate limiting provided by max_api_capacity should work as expected for all resources.

### Actual Behavior

It will not work as expected for some resources, [because of the reOktaID regex being used to normalize endpoint URLs](https://github.com/okta/terraform-provider-okta/blob/1f4d795339d011f61fd383cf9cb794e2bac39805/okta/internal/apimutex/apimutex.go#L132-L140), that will also replace path segments with 20 or more chars, leading to unrecognized buckets, and defaulting to root bucket "/":

```go
var (
	reOktaID = regexp.MustCompile(`[\w]{20}`)
)

func (m *APIMutex) get(method, endPoint string) *APIStatus {
	// the important point here is the replace all is performing this
	// transformation for the bucket lookup /api/v1/users/abcdefghij0123456789
	// to /api/v1/users/ID
	key := m.normalizedKey(method, path)
	bucket, ok := m.buckets[key]
	if !ok {
		return m.status["/"]
	}
	return m.status[bucket]
}
```

The test below shows the extent of the issue:

```go
func TestInvalidBuckets(t *testing.T) {
	reOktaID = regexp.MustCompile(`[\w]{20}`)

	uniquePaths := getUniquePaths()
	invalidBuckets := make(map[string][]string, 0)
	for path, bucket := range uniquePaths {
		submatch := reOktaID.FindString(path)

		if submatch != "" {
			if _, exists := invalidBuckets[bucket]; !exists {
				invalidBuckets[bucket] = make([]string, 0)
			}
			invalidBuckets[bucket] = append(invalidBuckets[bucket], path)
			t.Logf("%s has a path segment %s with 20 or more chars, bucket %s will default to '/' instead",
				path, submatch, bucket)
			t.Fail()
		}
	}
	prettyPrint, _ := json.MarshalIndent(invalidBuckets, "", "  ")
	t.Logf("invalid buckets: %s\n", string(prettyPrint))
}

func getUniquePaths() map[string]string {
	uniquePathMap := make(map[string]string, 0)
	for _, line := range rateLimitLines {
		split := strings.Split(line, " ")
		path := split[0]               //path
		uniquePathMap[path] = split[2] //bucket
	}
	return uniquePathMap
}
```
The following bucket / endpoint pairs will default to the root bucket "/" instead of their intended one. I did not check which endpoints are actually used by Terraform Okta resources, but I assume at least some of them are.

```json
{
  "/api/v1": [
    "/api/v1/authorizationServers/ID/policies/ID",
    "/api/v1/authorizationServers/ID/claims/ID",
    "/api/v1/authorizationServers/ID/credentials/keys",
    "/api/v1/authorizationServers/ID/lifecycle/activate",
    "/api/v1/authorizationServers/ID/claims",
    "/api/v1/authorizationServers/ID/scopes/ID",
    "/api/v1/authorizationServers/ID/apiResources/ID/scopes",
    "/api/v1/authorizationServers/ID/apiResources/ID",
    "/api/v1/authorizationServers/ID/policies/ID/rules/ID/lifecycle/deactivate",
    "/api/v1/productFeatures/earlyAccessAutoEnroll",
    "/api/v1/authorizationServers/ID/clients",
    "/api/v1/authorizationServers/ID/credentials/lifecycle/keyRotate",
    "/api/v1/authorizationServers/ID/apiResources",
    "/api/v1/authorizationServers/ID/clients/ID/tokens",
    "/api/v1/authorizationServers/ID/policies/ID/lifecycle/activate",
    "/api/v1/authorizationServers",
    "/api/v1/authorizationServers/ID/clients/ID/tokens/ID",
    "/api/v1/authorizationServers/ID",
    "/api/v1/authorizationServers/ID/lifecycle/deactivate",
    "/api/v1/authorizationServers/ID/credentials/keys/ID",
    "/api/v1/authorizationServers/ID/policies/ID/clients",
    "/api/v1/authorizationServers/ID/policies/ID/rules/ID",
    "/api/v1/authorizationServers/ID/apiResources/ID/claims/ID",
    "/api/v1/authorizationServers/ID/policies/ID/lifecycle/deactivate",
    "/api/v1/authorizationServers/ID/apiResources/ID/claims",
    "/api/v1/authorizationServers/ID/apiResources/ID/scopes/ID",
    "/api/v1/authorizationServers/ID/policies/ID/rules",
    "/api/v1/authorizationServers/ID/policies/ID/rules/ID/lifecycle/activate",
    "/api/v1/authorizationServers/ID/policies",
    "/api/v1/authorizationServers/ID/scopes"
  ],
  "/api/v1/certificateAuthorities": [
    "/api/v1/certificateAuthorities",
    "/api/v1/certificateAuthorities/ID/registrationAuthorities/ID/resetPassword",
    "/api/v1/certificateAuthorities/ID",
    "/api/v1/certificateAuthorities/ID/cert",
    "/api/v1/certificateAuthorities/ID/registrationAuthorities/ID/challenge",
    "/api/v1/certificateAuthorities/ID/registrationAuthorities/ID",
    "/api/v1/certificateAuthorities/bootstrap",
    "/api/v1/certificateAuthorities/ID/registrationAuthorities"
  ],
  "/api/v1/users": [
    "/api/v1/users/ID/lifecycle/expire_password_with_temp_password",
    "/api/v1/users/ID/authenticatorEnrollments",
    "/api/v1/users/me/credentials/change_recovery_question",
    "/api/v1/users/ID/credentials/change_recovery_question",
    "/api/v1/users/ID/authenticatorEnrollments/ID",
    "/api/v1/users/ID/credentials/forgot_password_recovery_question"
  ]
}
```

This issue has impact only if multiple resources incorrectly targeting the root bucket are used at the same time.
In that case, the quota for multiple resources will be mixed up, leading to potential rate limit errors.

### References

- The regex was introduced by #1356

**Comment by monde** [comment]
Thanks for all the details @clementdenis , I consider this a regression. I'll try and get a fix released today. `ReplaceAllStringFunc` will be my friend here to do a skip list of 20 character path elements that shouldn't be normalized.

```go
	path = reOktaID.ReplaceAllStringFunc(endPoint, func(what string) string {
		if what == "authorizationServers" {
			return what
		}
		return "ID"
	})
```

**Comment by monde** [comment]
Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-565039

**Comment by monde** [comment]
Fix released in v3.40.0 https://github.com/okta/terraform-provider-okta/releases/tag/v3.40.0
